src/bun.js/api/bun/subprocess.zig

```diff
@@ -1831,6 +1867,8 @@ pub const Subprocess = struct {
             }
         }
 
+        should_close_memfd = false;
```

Comment:
this variable seems to only be used once. Replace it with `Environment.isLinux`?

Reply:
it's used for errors, to ensure we close the memfd by default unless there's an error